src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -242,6 +297,140 @@ private static string ValueToHexString(object value)
             return result;
         }
 
+        private static bool InternalFlagsFormat(EnumInfo enumInfo, ulong resultValue, Span<char> destination, out int charsWritten, out bool isDestinationTooSmall)
```

Comment:
There is a lot of code duplication between the new generic and the existing non-generic paths. Can this be refactored to avoid this duplication?

Reply:
Working on this now, I will add a new commit soon

Reply:
@jkotas I've refactored the shared code. I've also updated the description with some more metrics. Perf and allocation characteristics remain the same after the refactor.